src/centengine/engine_impl.cc

```diff
@@ -753,6 +754,127 @@ grpc::Status engine_impl::DeleteServiceDowntime(grpc::ServerContext* context
   return grpc::Status::OK;
 }
 
+grpc::Status engine_impl::DeleteHostDowntimeFull(
+    grpc::ServerContext* context __attribute__((unused)),
+    const DowntimeCriterias* request,
+    CommandSuccess* response) {
+  auto fn = std::packaged_task<int32_t(void)>([request]() -> int32_t {
+    downtime::type downtime_type = downtime::host_downtime;
+    std::list<std::shared_ptr<com::centreon::engine::downtimes::downtime>>
+        dtlist;
+    std::cout << "has start" << request->has_start() << std::endl;
+    for (auto it = downtimes::downtime_manager::instance()
+                       .get_scheduled_downtimes()
+                       .begin(),
+              end = downtimes::downtime_manager::instance()
+                        .get_scheduled_downtimes()
+                        .end();
+         it != end; ++it) {
+      auto dt = it->second;
+
+      if (!(request->host_name().empty()) &&
+          (dt->get_hostname() != request->host_name()))
+        continue;
+      if ((request->has_start()) &&
+          (dt->get_start_time() != request->start().value()))
+        continue;
+      if ((request->has_end()) &&
+          (dt->get_end_time() != request->end().value()))
+        continue;
+      if ((request->has_fixed()) &&
+          (dt->is_fixed() != request->fixed().value()))
+        continue;
+      if ((request->has_triggered_by()) &&
+          (dt->get_triggered_by() != request->triggered_by().value()))
+        continue;
+      if ((request->has_duration()) &&
+          (dt->get_duration() != request->duration().value()))
+        continue;
+      if (!(request->author().empty()) &&
+          (dt->get_author() != request->author()))
+        continue;
+      if (!(request->comment_data().empty()) &&
+          (dt->get_comment() != request->comment_data()))
+        continue;
+      dtlist.push_back(dt);
```

Comment:
why so many parenthesis?
if (request->has_duration() && ... is suffisant.
idem around !=, no need of ()